File: UAS_OOP/soal_4_BankAccount.cpp
class BankAccount:
    def __init__(self, owner, balance=0):
        self.owner = owner
        self.balance = balance

    def deposit(self, amount):
        # Metode untuk melakukan penyetoran uang
        if amount > 0:
            self.balance += amount
            print(f"{amount} berhasil disetor. Saldo terkini: {self.balance}")
        else:
            print("Jumlah setoran harus lebih dari 0.")

    def withdraw(self, amount):
        # Metode untuk melakukan penarikan uang
        if 0 < amount <= self.balance:
            self.balance -= amount
            print(f"{amount} berhasil ditarik. Saldo terkini: {self.balance}")
        elif amount > self.balance:
            print("Saldo tidak mencukupi untuk melakukan penarikan.")
        else:
            print("Jumlah penarikan harus lebih dari 0.")

    def display_balance(self):
        # Metode untuk menampilkan saldo terkini
        print(f"Saldo terkini untuk pemilik {self.owner}: {self.balance}")

# Membuat objek rekening bank
my_account = BankAccount("John Doe", balance=1000)

# Menampilkan saldo awal
my_account.display_balance()

# Melakukan beberapa penyetoran dan penarikan uang
my_account.deposit(500)
my_account.withdraw(200)
my_account.deposit(1000)

# Menampilkan saldo terkini
my_account.display_balance()

File: RALAT_UAS_OOP/soal_4_BankAccount.php
<?php
class BankAccount {
    public $owner;
    public $balance;

    function __construct($owner, $balance = 0) {
        $this->owner = $owner;
        $this->balance = $balance;
    }

    function deposit($amount) {
        if ($amount > 0) {
            $this->balance += $amount;
            echo "$amount berhasil disetor. Saldo terkini: $this->balance<br>";
        } else {
            echo "Jumlah setoran harus lebih dari 0.<br>";
        }
    }

    function withdraw($amount) {
        if ($amount > 0 && $amount <= $this->balance) {
            $this->balance -= $amount;
            echo "$amount berhasil ditarik. Saldo terkini: $this->balance<br>";
        } elseif ($amount > $this->balance) {
            echo "Saldo tidak mencukupi untuk melakukan penarikan.<br>";
        } else {
            echo "Jumlah penarikan harus lebih dari 0.<br>";
        }
    }

    function displayBalance() {
        echo "Saldo terkini untuk pemilik $this->owner: $this->balance<br>";
    }
}

// Membuat objek rekening bank
$myAccount = new BankAccount("John Doe", 1000);

// Menampilkan saldo awal
$myAccount->displayBalance();

// Melakukan beberapa penyetoran dan penarikan uang
$myAccount->deposit(500);
$myAccount->withdraw(200);
$myAccount->deposit(1000);

// Menampilkan saldo terkini
$myAccount->displayBalance();
?>
